examples/blaze/src/main/scala/com/example/http4s/blaze/demo/server/service/FileService.scala

```diff
@@ -0,0 +1,51 @@
+package com.example.http4s.blaze.demo.server.service
+
+import java.io.{File, FileOutputStream, OutputStream}
+
+import cats.effect.Effect
+import cats.syntax.flatMap._
+import cats.syntax.functor._
+import com.example.http4s.blaze.demo.StreamUtils
+import fs2.Stream
+import org.http4s.multipart.Part
+
+import scala.concurrent.ExecutionContext.Implicits.global
+
+class FileService[F[_]](implicit F: Effect[F], S: StreamUtils[F]) {
+
+  def homeDirectories(depth: Option[Int]): Stream[F, String] =
+    S.env("HOME").flatMap { maybePath =>
+      val ifEmpty = S.error("HOME environment variable not found!")
+      maybePath.fold(ifEmpty)(directories(_, depth.getOrElse(1)))
+    }
+
+  def directories(path: String, depth: Int): Stream[F, String] = {
+
+    def dir(f: File, d: Int): Stream[F, File] = {
+      val dirs = Stream.emits(f.listFiles().toSeq).filter(_.isDirectory).covary[F]
+
+      if (d <= 0) Stream.empty
+      else if (d == 1) dirs
+      else dirs ++ dirs.flatMap(x => dir(x, d - 1))
+    }
+
+    S.evalF(new File(path)).flatMap { file =>
+      dir(file, depth)
+        .map(_.getName)
+        .filter(!_.startsWith("."))
+        .intersperse("\n")
+    }
+  }
+
+  def store(part: Part[F]): Stream[F, Unit] = {
+    val os: F[OutputStream] =
+      for {
+        home      <- F.delay(sys.env.getOrElse("HOME", "/tmp"))
+        filename  <- F.delay(part.filename.getOrElse("sample"))
+        output    <- F.delay(new FileOutputStream(s"$home/$filename"))
```

Comment:
In this case, I think `fs2.io.file` helpers are more brief than encouraging people to use output streams directly.

Reply:
Actually I'm using part of the `fs2.io` api below but I'm not sure this code can be replaced by any of the `fs2.io.file`. If you know how to I'd be glad to change this.

Reply:
Nevermind @jmcardon I found the clean way to do this, thanks for putting pressure on me :D

```scala
for {
      home      <- S.evalF(sys.env.getOrElse("HOME", "/tmp"))
      filename  <- S.evalF(part.filename.getOrElse("sample"))
      path      <- S.evalF(Paths.get(s"$home/$filename"))
      _         <- part.body to fs2.io.file.writeAll(path)
    } yield ()
```

Reply:
no problem, good job :+1: